src/main/java/duke/DukeCommandMatcher.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import duke.exceptions.CommandNotFoundException;
+import duke.exceptions.DateFormatException;
+import duke.exceptions.LackOfTimeException;
+import duke.exceptions.NullCommandContentException;
+import duke.exceptions.NullCommandException;
+import duke.exceptions.TaskNotSpecifyException;
+import duke.exceptions.TaskOutOfBoundException;
+import duke.storage.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.SingletonTaskList;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.utils.Constants;
+import duke.utils.UtilFunction;
+
+
+/**
+ * The class to match the command to Duke's response.
+ */
+public class DukeCommandMatcher {
+
+    private static final List<String> commandList = new ArrayList<>(Arrays.asList(Constants.LISTPATTERN,
```

Comment:
this constant should be in all caps! `COMMAND_LIST`

Reply:
Ahh, forgot about it, thanks.